### PR TITLE
Fix error loading theme configuration on PHP 7.2

### PR DESCRIPTION
**Pull request by Alanaktion**
This fixes an issue loading theme configuration on PHP 7.2 by checking that `$parentPathPieces` is an array before counting it, avoiding an error when the value is `NULL`. PHP [no longer allows](http://php.net/manual/en/migration72.incompatible.php#migration72.incompatible.warn-on-non-countable-types) running `count()` on non-countable types.

An easy way to trigger this bug in the current 2.2 branch is to run the `setup:upgrade` command:
```
$ php bin/magento setup:upgrade
...
Running data recurring...
Warning: count(): Parameter must be an array or an object that implements Countable in vendor/magento/framework/View/Design/Theme/ThemeList.php on line 237
```

This might be better to merge into 2.3, but I'm not sure what the planned version support/release scheduled for 2.2 vs 2.3 is.

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=12606) <br/>All committers have signed the CLA.

**Comment by avoelkl** [comment]
@Alanaktion Even though this helps PHP 7.2 compatibility, performing the additional `is_array` check is good anyway so i think it's ok go into 2.2 :)
It can be forward ported to 2.3 after we merged this one (either you provide an additional PR or the core team can do it). First we process this one.

**Review by avoelkl — APPROVED** [review]

**Comment by avoelkl** [comment]
@Alanaktion Thanks for the PR!

**Pull request merged**
